gcp/api/v1/osv_service_v1.proto

```diff
@@ -67,6 +67,49 @@ message QueryAffectedBatchParameters {
   BatchQuery query = 1;
 }
 
+// Parameters for DetermineVersion.
+message DetermineVersionParameters {
+  VersionQuery query = 1;
+}
+
+// The version query.
+message VersionQuery {
+  // The name of the dependency. Can be empty.
+  string name = 1;
+  repeated FileHash file_hashes = 2;
+}
+
+// Information about the files in the repository
+// to identify the version.
+message FileHash {
+  // The file path inside the repository. 
+  // Can be empty but providing it increases matching quality.
+  string file_path = 1;
+  string hash_type = 2;
+  repeated bytes hash = 3;
+}
+
+// Result of DetmineVersion.
+message VersionMatchList {
+  repeated VersionMatch matches = 1;
+}
+
+// Match information for the provided VersionQuery.
+message VersionMatch {
+  enum IDType {
+    UNSPECIFIED = 0;
+    OSV = 1;
+    CPE = 2;
+    VERSION = 3;
```

Comment:
The idea was to have a single version type instead of providing different ones based on the type. So, a GIT type returns repository information, OSV an OSV based identifier (I think I mixed this up with OSV vuln ids) and CPE the full cpe tag.

I can see it working the way you suggest it but we need to change the data collection slightly because we don't store the tag in datastore.

Reply:
Right, I think we should store the original tag either way. It will give us more flexibility if we need to change our version parsing algorithm/regex too.